File: LeetCode/Python Solutions/Design a Stack With Increment Operation/design.py
class CustomStack:
    def __init__(self, maxSize: int):
        self.stack = []
        self.inc = []  # Renamed from 'increment' to 'inc'
        self.maxSize = maxSize

    def push(self, x: int) -> None:
        if len(self.stack) < self.maxSize:
            self.stack.append(x)
            self.inc.append(0)

    def pop(self) -> int:
        if not self.stack:
            return -1
        if len(self.stack) > 1:
            self.inc[-2] += self.inc[-1]
        return self.stack.pop() + self.inc.pop()

    def increment(self, k: int, val: int) -> None:
        i = min(k - 1, len(self.stack) - 1)
        if i >= 0:
            self.inc[i] += val

File: LeetCode/Java Solutions/Design a Stack With Increment Operation/design.java
class CustomStack {
    private int[] stack;
    private int[] increment;
    private int top;
    private int maxSize;

    public CustomStack(int maxSize) {
        this.stack = new int[maxSize];
        this.increment = new int[maxSize];
        this.top = -1;
        this.maxSize = maxSize;
    }
    
    public void push(int x) {
        if (top < maxSize - 1) {
            top++;
            stack[top] = x;
        }
    }
    
    public int pop() {
        if (top == -1) {
            return -1;
        }
        int result = stack[top] + increment[top];
        if (top > 0) {
            increment[top - 1] += increment[top];
        }
        increment[top] = 0;
        top--;
        return result;
    }
    
    public void increment(int k, int val) {
        int i = Math.min(k - 1, top);
        if (i >= 0) {
            increment[i] += val;
        }
    }
}